src/plugins/management/public/components/management_sidebar_nav/management_sidebar_nav.tsx

```diff
@@ -17,184 +17,98 @@
-    ? section.euiIconType
-    : section.icon
-    ? section.icon
-    : 'empty';
-
-  return {
-    icon: <EuiIcon type={iconType} size="m" />,
-    ...managementSectionOrAppToNav(section),
-  };
-};
-
-const managementAppToNavItem = (selectedId?: string, parentId?: string) => (
-  app: ManagementApp
-) => ({
-  isSelected: selectedId === app.id,
-  href: `#/management/${parentId}/${app.id}`,
-  ...managementSectionOrAppToNav(app),
+const navMenuLabel = i18n.translate('management.nav.menu', {
+  defaultMessage: 'Management menu',
 });
 
-const legacySectionToNavSection = (section: LegacySection) => ({
-  name: section.display,
-  id: section.id,
-  icon: section.icon ? <EuiIcon type={section.icon} /> : null,
-  items: [],
-  'data-test-subj': section.id,
-  // @ts-ignore
-  order: section.order,
-});
-
-const legacyAppToNavItem = (app: LegacyApp, selectedId: string) => ({
-  isSelected: selectedId === app.id,
-  name: app.display,
-  id: app.id,
-  href: app.url,
-  'data-test-subj': app.id,
-  // @ts-ignore
-  order: app.order,
-});
-
-const sectionVisible = (section: LegacySection | LegacyApp) => !section.disabled && section.visible;
-
-const sideNavItems = (sections: ManagementSection[], selectedId: string) =>
-  sections.map(section => ({
-    items: section.getAppsEnabled().map(managementAppToNavItem(selectedId, section.id)),
-    ...managementSectionToNavSection(section),
-  }));
-
-const findOrAddSection = (navItems: NavSection[], legacySection: LegacySection): NavSection => {
-  const foundSection = navItems.find(sec => sec.id === legacySection.id);
-
-  if (foundSection) {
-    return foundSection;
-  } else {
-    const newSection = legacySectionToNavSection(legacySection);
-    navItems.push(newSection);
-    navItems.sort((a: NavSection, b: NavSection) => a.order - b.order); // only needed while merging platform and legacy
-    return newSection;
-  }
-};
-
-export const mergeLegacyItems = (
-  navItems: NavSection[],
-  legacySections: LegacySection[],
-  selectedId: string
-) => {
-  const filteredLegacySections = legacySections
-    .filter(sectionVisible)
-    .filter(section => section.visibleItems.length);
-
-  filteredLegacySections.forEach(legacySection => {
-    const section = findOrAddSection(navItems, legacySection);
-    legacySection.visibleItems.forEach(app => {
-      section.items.push(legacyAppToNavItem(app, selectedId));
-      return section.items.sort((a, b) => a.order - b.order);
-    });
-  });
-
-  return navItems;
-};
+/** @internal **/
+export const ManagementSidebarNav = ({
+  selectedId,
+  sections,
+  history,
+}: ManagementSidebarNavProps) => {
+  const HEADER_ID = 'stack-management-nav-header';
+  const [isSideNavOpenOnMobile, setIsSideNavOpenOnMobile] = useState(false);
+  const toggleOpenOnMobile = () => setIsSideNavOpenOnMobile(!isSideNavOpenOnMobile);
+
+  const sectionsToNavItems = (managementSections: ManagementSection[]) => {
+    if (!managementSections) {
```

Comment:
Is it necessary to perform this check? Seems like this should always be an array.